Source/JavaScriptCore/bytecode/PutByStatus.cpp

```diff
@@ -394,6 +395,10 @@ PutByStatus PutByStatus::computeFor(JSGlobalObject* globalObject, const Structur
         // we don't want to be adding properties to strings.
         if (!structure->typeInfo().isObject())
             return PutByStatus(LikelyTakesSlowPath);
+
+        // If the structure is for prototype, we should do a slow path which can invalidate MegamorphicCache.
+        if (structure->mayBePrototype())
+            return PutByStatus(LikelyTakesSlowPath);
```

Comment:
Hmm, so previously we could have had a prototype set be cached + inlined, but not anymore? I guess this happens pretty infrequently

Reply:
This happens very infrequently, and it didn't cause any performance issues (I isolated this part in a separate A/B test and ensured).